client/app/(auth)/createTeam.tsx

```diff
@@ -0,0 +1,364 @@
+    control,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+    reset,
+  } = useForm<TeamEditModel>({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      abbreviation: "",
+      name: "",
+      logoUrl: "",
+    },
+    mode: "onSubmit",
+  });
+
+  useEffect(() => {
+    const setDefaultTeamData = async () => {
+      if (currentUser) {
+        const teams = await UserController.getUserTeams(currentUser.uid);
+
+        // A user should only have either 0 or 1 team on the createTeam screen.
+        if (teams?.length === 1) {
+          const matchedLogo = logoOptions.find(
+            (option) => option.url === teams[0]!.logoUrl,
+          );
+          if (matchedLogo) {
+            setSelectedLogoOption(matchedLogo);
+          }
+
+          reset({
+            abbreviation: teams[0].abbreviation,
+            name: teams[0].name,
+            logoUrl: teams[0].logoUrl,
+          });
+        }
+      }
+    };
+
+    setDefaultTeamData();
+  }, [currentUser, reset]);
+
+  const handleCreateTeam = async (model: TeamEditModel) => {
+    setIsLoading(true);
+    try {
+      if (!currentUser) {
+        router.push("/(auth)");
+        return;
+      }
+
+      const userId = currentUser.uid;
+      const teams = await UserController.getUserTeams(userId);
+
+      let teamId: string;
+
+      if (teams?.length === 0) {
+        teamId = await UserController.addUserTeam(userId, model);
+      } else if (teams?.length === 1) {
+        teamId = teams[0].id;
+        await UserController.editUserTeam(userId, teamId, model);
+      } else {
+        throw new Error("Unexpected number of teams in create team screen");
+      }
+
+      const teamData = await UserController.getUserTeam(userId, teamId);
+      if (teamData) {
+        dispatch(
+          setTeam({
+            id: teamData.id,
+            name: teamData.name,
+            abbreviation: teamData.abbreviation,
+            logoUrl: teamData.logoUrl,
+          }),
+        );
+      }
+
+      router.replace("/(protected)");
+    } catch (error) {
+      console.error("Failed to create team:", error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <View className="flex-1 bg-gray-950">
+      <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+        <View className="flex-1">
+          <KeyboardAvoidingView behavior="padding" className="flex-1">
+            <View className="flex-1 flex-col px-6 py-4">
+              <View className="my-8 size-8 self-end" />
+
+              <View className="flex-row items-start">
+                <Pressable
+                  className="mr-3"
+                  onPress={() => router.push("/(auth)/createProfile")}
+                >
+                  <BackIcon fill="white" height={25} width={25} />
+                </Pressable>
+                <Text className="pbk-h5 mb-8 text-base-white">
+                  Create your team
+                </Text>
+              </View>
```

Comment:
To remove the spacer and fix up the icon and title spacing, and replace the arrow svg with a icon from phosphor:
```suggestion
              <View className="mb-8 mt-24 flex-row items-center gap-4">
                <Pressable
                  className="size-8 items-center justify-center rounded-md border border-gray-900 p-4"
                  onPress={() => router.push("/(auth)/createProfile")}
                >
                  <ArrowLeft color="white" size={20} weight="bold" />
                </Pressable>
                <Text className="pbk-h5 text-base-white">Create your team</Text>
              </View>
```

Reply:
Cool, this worked great, thanks!